lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
-          return false if id.nil?
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          api_session.modify_identifier(id, _status: IDENTIFIER_STATUS[:unavailable])
+        def map_metadata(hyacinth_metadata, datacite_data, target_url)
+          return unless hyacinth_metadata
+          # BEGIN metadata required by DataCite
+          # Of course, we can come up with other/better default values
+          datacite_data.title = hyacinth_metadata.title || 'Placeholder Title'
+          datacite_data.creators = hyacinth_metadata.creators || 'Placeholder Creator'
+          map_publication_metadata(hyacinth_metadata, datacite_data)
+          datacite_data.resource_type_general = hyacinth_metadata.type_of_resource || "Text"
+          datacite_data.url = target_url || 'https://library.columbia.edu'
+          # END metadata required by DataCite
         end
 
-        private
+        def map_publication_metadata(hyacinth_metadata, datacite_data)
+          datacite_data.publisher = hyacinth_metadata.publisher || "Placeholder Publisher"
+          datacite_data.publication_year = hyacinth_metadata.date_issued_start_year.to_i ||
+                                           "Placeholder Publication Year"
+        end
 
-          # setup EZID API info: credentials, url, etc.
-          def api_session
-            Thread.current[:datacite_ezid_api_session] ||= Datacite::EzidSession.new(DATACITE[:user], DATACITE[:password])
-          end
+        # Returns true if an identifier exists in the external management system
+        def exists?(doi)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          resp = rest_api.get_dois(doi)
+          rest_api.parse_doi_from_api_response_body(resp.body).present?
+        end
 
-          def http_error_message(response)
-            return "" if response.nil?
-            "HTTP status code: #{response.http_status_code}\n" \
-            "HTTP server message: #{response.http_server_message}\n" \
-            "response body: #{response.body}"
-          end
+        # @param id [String]
+        # @param digital_object [DigitalObject]
+        # @param location_uri [String]
+        # @return [Boolean] true if this adapter can handle this type of identifier
+        def update_impl(_id, _digital_object, _location_uri)
+          # needs state as a parameter in order to be able to update DOI state
+          # also, prefer a Metadata as an argument instead of a Digital Object. Metadata offers "accessors" to
+          # get to metadata inside Digital Object
+          raise NotImplementedError
+        end
 
-          # TODO: Remove this method when there's a general logging proxy
-          def log(status, message); end
+        def tombstone_impl(_id)
+          # switch state to registered?
+          raise NotImplementedError
+        end
```

Comment:
It looks like you removed the old tombstone_impl but still need to add the new version.

Reply:
I'm asking that we replace this `raise` with a TODO and take this up in HYACINTH-876

Reply:
(closely related to HYACINTH-393)